### PR TITLE
Try: Remove menu item min-width.

### PR DESCRIPTION
**Pull request by jasmussen**
Popovers have a min-width. Which means you get very wide popovers in some cases:

Screenshot 2020-09-10 at 13 02 26

This PR tries removing that, so they look just right (sized according to the content inside):

Screenshot 2020-09-10 at 13 01 40

So why was it added in the first place? Well, it was added back in #9086 because an auto-expanding URL field didn't resize it. With this PR I'm revisiting that, though, because: should a textfield ever _expand_ a popover? Isn't it best those have fixed sizes? And currently, they do:

Screenshot 2020-09-10 at 13 01 47

So, this PR would be a nice little win for toolbar dropdowns, and even a bit of cleanup. But it'd be good to test existing popovers to make sure they work as intended.

### CODE REVIEW

**Comment by ItsJonQ** [comment]
@jasmussen Woo! Thanks for this one ❤️ 

I just tested it locally. I've noticed some issues with some of the Toolbar dropdowns (for the Cover block):

Screen Shot 2020-09-16 at 8 57 52 AM

Screen Shot 2020-09-16 at 8 57 57 AM

This could perhaps be resolved by adding `white-space: nowrap` on menu items/buttons (which they perhaps should have to begin with 🤔 )

This leads to another issue, at least if the dropdown happens to have overflow 🙈 :

Screen Shot 2020-09-16 at 9 02 01 AM


I gave that a shot. It worked (at establishing a better min-width). The other issue is that the icon is really close to the text (almost touching). This echos an issue with the Button/MenuItem icon setup, rather than min-width.

Screen Shot 2020-09-16 at 8 58 53 AM

With all of that being said!! (Phew).
I don't think removing `min-width` is a causing these issues. If anything, I think it reveals some underlying UI issues.

(Not really sure how deep the rabbit hole goes on this one, but maybe worth taking a look?).

Ultimately though, I agree with your original mission in that, I too believe Dropdown menu widths should be more adaptive/flexible.

**Comment by jasmussen** [comment]
Excellent sleuthing, that was really helpful! I found a hardcoded 200px somewhere, which I changed to a min-width. Now it works:

Screenshot 2020-09-17 at 11 44 16

But now this one is a bit out there, so I'll look into that now :D 

Screenshot 2020-09-17 at 11 44 16

**Comment by jasmussen** [comment]
Got the ellipsis menu working, alas it required another wrapper around menu items, but I think that's okay.

Screenshot 2020-09-17 at 11 56 07

**Comment by jasmussen** [comment]
Okay I think I have it. 

Screenshot 2020-09-17 at 11 58 30

Screenshot 2020-09-17 at 12 00 21

Screenshot 2020-09-17 at 12 01 16

Screenshot 2020-09-17 at 12 01 23

**Comment by ItsJonQ** [comment]
@jasmussen  🎉 ! Looks like it's working :).
The test suites are a bit unhappy... most likely due to the new added selector.
Going to poke at those to see if I can make them happy again.

**Comment by ItsJonQ** [comment]
🙈 
I'm struggling with running E2E tests locally. For some reason, this is happening:

```
console.error
    Failed to load resource: net::ERR_PROXY_CONNECTION_FAILED

      170 |
      171 | 		// eslint-disable-next-line no-console
    > 172 | 		console[ logFunction ]( text );
          | 		^
      173 | 	} );
      174 | }
      175 |

      at CustomConsole.<anonymous> (../../node_modules/jest-environment-node/node_modules/jest-mock/build/index.js:814:25)
      at Page.<anonymous> (config/setup-test-framework.js:172:3)
      at Page._onLogEntryAdded (../../node_modules/puppeteer/lib/Page.js:203:18)
```

When I try to run 
```
npm run test-e2e:watch -- packages/e2e-tests/specs/editor/various/reusable-blocks.test.js
```

(I tried blowing away Docker's resources and starting from scratch. It didn't seem to help)

Don't let me be a blocker for this one! If anyone else can help with E2E tests, by all means ❤️

**Comment by jasmussen** [comment]
Thanks so much for looking and fixing! I'll have another stab, and otherwise we'll find a way.

**Comment by jasmussen** [comment]
I pushed a few tweaks and rebased/squashed. Let's see if it does anything.

**Comment by gziolo** [comment]
It looks like some e2e tests that use XPath need to be updated to reflect changes in HTML syntax.

File: package-lock.json [hunk 1]
```diff
@@ -46068,7 +46068,7 @@
 			"dependencies": {
 				"clone-deep": {
 					"version": "0.2.4",
-					"resolved": "https://registry.npmjs.org/clone-deep/-/clone-deep-0.2.4.tgz",
+					"resolved": "http://registry.npmjs.org/clone-deep/-/clone-deep-0.2.4.tgz",
```
Comment (gziolo) [inline package-lock.json:46071]: I guess it shouldn't be in this PR. We can clean it up later...

**Comment by gziolo** [comment]
I see similar e2e test failures in the `master` branch. It looks like it shouldn't block this PR.

**Comment by jasmussen** [comment]
Since this PR is not urgent, I'll let it sit until the failures in the master branch are taken care of, then I'll rebase and we can review again. Thanks so much for all the help @gziolo !

**Review by mkaz — APPROVED** [review]
Tests are now passing and works as described. 👍 

I noticed the properly sized drop down after applying the patch and did not find any regressions.

**Comment by jasmussen** [comment]
Awesome, let's try this, and I'll follow up and re-apply min-widths in specific cases if they turn out to be necessary!

**Pull request merged**